### PR TITLE
Updated api to create new opportunity

### LINKED ISSUE #229: Update API to create a new opportunity
Update the API in all sections of PostOpportunity with: https://opportunity-calendar.herokuapp.com/opportunity

Example Schema:
{
  "opportunityTitle": "string",
  "opportunityType": "string",
  "opportunityOrganisation": "string",
  "opportunityLocation": "string",
  "opportunityDescription": "string",
  "opportunityEligibility": "string",
  "opportunityRegistrationDeadline": "2021-05-02",
  "opportunityDate": "2021-05-02",
  "opportunityURL": "string",
  "onlyForFemale": true,
  "organisationLogoURL": "string"
}

### PR DESCRIPTION
**Pull request by Samridhi-98**
# Description

Updated API in all sections of the post opportunity and code refactoring.

Fixed #229 

## Type of change

Please check options that are relevant to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] I have pulled the latest changes in the repository
- [x] I have squashed my commits
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@vaishali614  @Manvityagi  Please review my PR. Also, there is no field for opportunityImage in the mongoose schema but the form section has the state property for the image. Should I remove it from the frontend part or simply open an issue in the backend repository to update the opportunity schema.

**Comment by vaishali614** [comment]
> @vaishali614 @Manvityagi Please review my PR. Also, there is no field for opportunityImage in the mongoose schema but the form section has the state property for the image. Should I remove it from the frontend part or simply open an issue in the backend repository to update the opportunity schema.

Please open an issue in backend repo. Other than that, your changes look good. When the field for image is added, please update your PR. Also, do take the latest pull from upstream before you update your PR.

**Comment by Manvityagi** [comment]
> @vaishali614 @Manvityagi Please review my PR. Also, there is no field for opportunityImage in the mongoose schema but the form section has the state property for the image. Should I remove it from the frontend part or simply open an issue in the backend repository to update the opportunity schema.

https://github.com/Girl-Code-It/Opportunity-Calendar-Backend/issues/65 
Have opened the issue in backend for the same

**Comment by Samridhi-98** [comment]
> > @vaishali614 @Manvityagi Please review my PR. Also, there is no field for opportunityImage in the mongoose schema but the form section has the state property for the image. Should I remove it from the frontend part or simply open an issue in the backend repository to update the opportunity schema.
> 
> [Girl-Code-It/Opportunity-Calendar-Backend#65](https://github.com/Girl-Code-It/Opportunity-Calendar-Backend/issues/65)
> Have opened the issue in backend for the same

Thank you, Will update this PR as soon as that PR will be merged, till the time I am keeping it as a draft PR.

**Comment by Manvityagi** [comment]
> > > @vaishali614 @Manvityagi Please review my PR. Also, there is no field for opportunityImage in the mongoose schema but the form section has the state property for the image. Should I remove it from the frontend part or simply open an issue in the backend repository to update the opportunity schema.
> > 
> > 
> > [Girl-Code-It/Opportunity-Calendar-Backend#65](https://github.com/Girl-Code-It/Opportunity-Calendar-Backend/issues/65)
> > Have opened the issue in backend for the same
> 
> Thank you, Will update this PR as soon as that PR will be merged, till the time I am keeping it as a draft PR.

Sounds good 👍

**Comment by Samridhi-98** [comment]
@vaishali614  @Manvityagi  I have updated for image URL please review.

**Comment by vaishali614** [comment]
> @vaishali614 @Manvityagi I have updated for image URL please review.

@Samridhi-98 I can't post an opportunity.
Screenshot (10)

CC: @Manvityagi

**Comment by Samridhi-98** [comment]
> > @vaishali614 @Manvityagi I have updated for image URL please review.
> 
> @Samridhi-98 I can't post an opportunity.
> Screenshot (10)
> 
> CC: @Manvityagi

@vaishali614 I am not sure but there is some error at the backend while testing the URL I am not able to retrieve anything, got this error message please let me know how can I fix this.

**Screenshot**
frontend

**Comment by Manvityagi** [comment]
> > > @vaishali614 @Manvityagi I have updated for image URL please review.
> > 
> > 
> > @Samridhi-98 I can't post an opportunity.
> > Screenshot (10)
> > CC: @Manvityagi
> 
> @vaishali614 I am not sure but there is some error at the backend while testing the URL I am not able to retrieve anything, got this error message please let me know how can I fix this.
> 
> **Screenshot**
> frontend

It should be working now, Please check

**Comment by Samridhi-98** [comment]
> > > > @vaishali614 @Manvityagi I have updated for image URL please review.
> > > 
> > > 
> > > @Samridhi-98 I can't post an opportunity.
> > > Screenshot (10)
> > > CC: @Manvityagi
> > 
> > 
> > @vaishali614 I am not sure but there is some error at the backend while testing the URL I am not able to retrieve anything, got this error message please let me know how can I fix this.
> > **Screenshot**
> > frontend
> 
> It should be working now, Please check

Yes, It's working.
@vaishali614  please review and let me know if I need to update anything.

**Comment by vaishali614** [comment]
@Samridhi-98 The changes look perfect. But, we've added two more fields in the schema: opportunityType, onlyForFemale.
Since you've been working on the opportunity form, can you add a dropdown for opportunityType with values JOB, INTERNSHIP, HACKATHON, SCHOLARSHIP, CONFERENCE, CODINGCOMPETITION and a toggle button for onlyForFemale in the form for these two fields? 
Apart from this, everything looks good.

You can refer to the Swagger Doc [here](https://opportunity-calendar.herokuapp.com/playground/#/default/post_opportunity)

**Comment by Manvityagi** [comment]
> @Samridhi-98 The changes look perfect. But, we've added two more fields in the schema: opportunityType, onlyForFemale.
> Since you've been working on the opportunity form, can you add a dropdown for opportunityType with values JOB, INTERNSHIP, HACKATHON, SCHOLARSHIP, CONFERENCE, CODINGCOMPETITION and a toggle button for onlyForFemale in the form for these two fields?
> Apart from this, everything looks good.
> 
> You can refer to the Swagger Doc [here](https://opportunity-calendar.herokuapp.com/playground/#/default/post_opportunity)

I think we can increase the level of this issue from 2 to 3, if you want to take this added task.

**Comment by Samridhi-98** [comment]
> @Samridhi-98 The changes look perfect. But, we've added two more fields in the schema: opportunityType, onlyForFemale.
> Since you've been working on the opportunity form, can you add a dropdown for opportunityType with values JOB, INTERNSHIP, HACKATHON, SCHOLARSHIP, CONFERENCE, CODINGCOMPETITION and a toggle button for onlyForFemale in the form for these two fields?
> Apart from this, everything looks good.
> 
> You can refer to the Swagger Doc [here](https://opportunity-calendar.herokuapp.com/playground/#/default/post_opportunity)

Sure, I will add it.

**Comment by Samridhi-98** [comment]
@vaishali614 Do I need to add a dropdown for opportunityType in each form, Since each and every opportunity type has its own separate form so it's pointless to add opportunityType dropdown.

**Comment by vaishali614** [comment]
> @vaishali614 Do I need to add a dropdown for opportunityType in each form, Since each and every opportunity type has its own separate form so it's pointless to add opportunityType dropdown.

Yeah I agree with you, that's why we're planning to reduce 6 forms into 1 form, so we'll need the dropdown. But for now, just add this dropdown in each form.

**Comment by Samridhi-98** [comment]
> > @vaishali614 Do I need to add a dropdown for opportunityType in each form, Since each and every opportunity type has its own separate form so it's pointless to add opportunityType dropdown.
> 
> Yeah I agree with you, that's why we're planning to reduce 6 forms into 1 form, so we'll need the dropdown. But for now, just add this dropdown in each form.

Okay.

**Comment by Samridhi-98** [comment]
@vaishali614  @Manvityagi  Please review.

**Comment by vaishali614** [comment]
1. You are using the same ID twice on the same page
2. There's some error on your end while making request. Please check.

Screenshot (53)

PS: Please make sure you test it completely before you update your PR for review.

**Comment by Samridhi-98** [comment]
> 1. You are using the same ID twice on the same page
> 2. There's some error on your end while making request. Please check.
> 
> Screenshot (53)
> 
> PS: Please make sure you test it completely before you update your PR for review.

@vaishali614  I am unable to send the request will you please help me out with this.

**Screeenshot**

main

ex1

ex2

File: src/Components/PostOpportunity/sections/FullTime/FullTimeForm.js [hunk 1]
```diff
@@ -63,20 +65,22 @@ class FullTimeForm extends Component {
...
+          opportunityType: this.state.type,
+          opportunityURL: this.state.jobURL,
+          opportunityTitle: this.state.title,
+          opportunityOrganisation: this.state.company,
+          organisationLogoURL: this.state.image,
+          opportunityDescription: this.state.jobDescription,
+          opportunityLocation: this.state.location,
+          opportunityEligibility: this.state.eligibility,
+          opportunityRegistrationDeadline: this.state.deadline,
+          opportunityOnlyForFemale: this.state.onlyForFemale,
```
Comment (vaishali614) [inline src/Components/PostOpportunity/sections/FullTime/FullTimeForm.js]: The field is 'onlyForFemale' not 'opportunityOnlyForFemale', 'opportunityId' field has been removed and there's one more field 'opportunityDate'.

opportunityDate can be same as deadline for jobs, internships, scholarships, but for others there should be a different input field.

File: src/Components/PostOpportunity/sections/FullTime/FullTimeForm.js [hunk 2]
```diff
@@ -170,6 +176,40 @@ class FullTimeForm extends Component {
                 {this.state.FieldEmptyError}
               </div>
 
+              <Form.Row>
+                <Form.Group as={Col} controlId="formGridType">
+                  <Form.Control
+                    as="select"
+                    size="lg"
+                    defaultValue="Type"
```
Comment (vaishali614) [inline src/Components/PostOpportunity/sections/FullTime/FullTimeForm.js]: Add 
name="type"
value={type}

File: src/Components/PostOpportunity/sections/FullTime/FullTimeForm.js [hunk 3]
```diff
@@ -170,6 +176,40 @@ class FullTimeForm extends Component {
...
+                    onChange={this.handleChange}
+                    style={{ marginTop: '30px' }}
+                  >
+                    <option defaultValue hidden>Opportunity type</option>
+                    <option value="Job">Job</option>
+                    <option value="Internship">Internship</option>
```
Comment (vaishali614) [inline src/Components/PostOpportunity/sections/FullTime/FullTimeForm.js]: values should be JOB, INTERNSHIP, HACKATHON, SCHOLARSHIP, CONFERENCE, CODINGCOMPETITION

**Review by vaishali614 — CHANGES_REQUESTED** [review]
- Please make sure these changes are done for all the forms.
- Do check the Swagger Doc to confirm the field names in case any errors are encountered.

**Comment by Samridhi-98** [comment]
@vaishali614  I have updated as you suggested and each form is properly working please review and let me know if anything else to update.

**Review by vaishali614 — APPROVED** [review]

**Pull request merged**